spatial/kdtree/kdtree_simple_example_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kdtree_test
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/spatial/kdtree"
+)
+
+func ExampleTree() {
+	// Example data from https://en.wikipedia.org/wiki/K-d_tree
+	points := kdtree.Points{{2, 3}, {5, 4}, {9, 6}, {4, 7}, {8, 1}, {7, 2}}
+
+	t := kdtree.New(points, false)
+	q := kdtree.Point{8, 7}
+	p, d := t.Nearest(q)
+	fmt.Printf("%v is closest point to %v, d=%f\n", p, q, math.Sqrt(d))
+	// Output:
+	// [9 6] is closest point to [8 7], d=1.414214
+}
+
+func ExampleTree_bounds() {
+	// Example data from https://en.wikipedia.org/wiki/K-d_tree
+	points := kdtree.Points{{2, 3}, {5, 4}, {9, 6}, {4, 7}, {8, 1}, {7, 2}}
+
+	t := kdtree.New(points, true)
+	fmt.Printf("Bounding box of points is %+v\n", t.Root.Bounding)
+	// Output:
+	// Bounding box of points is &{Min:[2 1] Max:[9 7]}
+}
+
+func ExampleTree_Do() {
+	// Example data from https://en.wikipedia.org/wiki/K-d_tree
+	points := kdtree.Points{{2, 3}, {5, 4}, {9, 6}, {4, 7}, {8, 1}, {7, 2}}
+
+	// Print all points in the data set within 3 of (3, 5).
+	t := kdtree.New(points, false)
+	q := kdtree.Point{3, 5}
+	t.Do(func(c kdtree.Comparable, _ *kdtree.Bounding, _ int) (done bool) {
+		// Compare each distance and output points
+		// with a distance less than 9. Distance
```

Comment:
`less than 3`, maybe also write `9` below as `3*3`?

Reply:
I've said "Euclidean distance less than 3" now. Formally, it's still a distance less than 9, but this is clearer now.

Reply:
Yes, I think it is clearer now.